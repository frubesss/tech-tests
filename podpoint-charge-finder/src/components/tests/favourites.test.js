import React from 'react';
import Favourites from '../Favourites'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme';

describe('Favourites Component', () => {

    let wrapper;

    const mockedFavourites = [{
        UUID: 1,
        AddressInfo: {
            Title: 'Albert Dock Road',
            AddressLine1: '22 Boaty Road',
            AddressLine2: 'Sealife Avenue',
            Postcode: 'L1 9EH',
            Distance: 0.126332312
        }
    },
        {
            UUID: 2,
            AddressInfo: {
                Title: 'Super Duper Pod Point',
                AddressLine1: 'Testing Avenue',
                AddressLine2: 'Wirral',
                Postcode: 'L12 2ER',
                Distance: 0.1236332312
            }
        }
    ];

    beforeEach(() => {
        wrapper = shallow(<Favourites favourites={mockedFavourites}/>);
    });

    it('Snapshot Test', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a list of pod point favourites', () => {
        expect(wrapper.find('.item-list').children()).toHaveLength(2);
    });

    it('should round up the distance to two decimal places with correct formatting', () => {
        expect(wrapper.find('.item-distance').first().text()).toEqual('0.13km');
    });

});
