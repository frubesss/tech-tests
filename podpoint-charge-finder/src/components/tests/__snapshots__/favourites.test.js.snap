// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favourites Component Snapshot Test 1`] = `
<div>
  <ul
    className="item-list"
  >
    <li
      className="item"
    >
      <i
        className="fa fa-map-marker map-marker"
      />
      <ul
        className="item-address"
      >
        <li>
          <a
            className="item-heading"
          >
            Albert Dock Road
          </a>
        </li>
        <li>
          <a
            className="item-subheading"
          >
            22 Boaty Road - L1 9EH
          </a>
        </li>
      </ul>
      <a
        className="item-distance"
      >
        0.13
        km
      </a>
    </li>
    <li
      className="item"
    >
      <i
        className="fa fa-map-marker map-marker"
      />
      <ul
        className="item-address"
      >
        <li>
          <a
            className="item-heading"
          >
            Super Duper Pod Point
          </a>
        </li>
        <li>
          <a
            className="item-subheading"
          >
            Testing Avenue - L12 2ER
          </a>
        </li>
      </ul>
      <a
        className="item-distance"
      >
        0.12
        km
      </a>
    </li>
  </ul>
</div>
`;
